spec/lib/system_commands_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'spec_helper'
+
+describe SystemCommands do
+
+  it 'should return error on timeout properly' do
+    expect {
+      sh!('sleep', 2, { timeout: 1 })
+    }.to raise_error
+  end
+
+  it 'should not return error without timeout set' do
```

Comment:
This seems unnececssary. I'd rather have a test that checks that an exception is raised if the command fails with a non-zero exit status.